src/test/java/com/salesforce/dynamodbv2/mt/mappers/MtAmazonDynamoDbStreamsBaseTest.java

```diff
@@ -266,13 +86,50 @@ private static void assertGetRecords(MtAmazonDynamoDbStreams streams, Collection
                 AmazonDynamoDbLocal.getAmazonDynamoDbStreamsLocal());
 
             return java.util.stream.Stream.of(
-                Arguments.of(indexMtDynamoDb, indexMtDynamoDbStreams),
-                Arguments.of(indexBinaryHkMtDynamoDb, indexBinaryHkMtDynamoDbStreams),
-                Arguments.of(tableMtDynamoDb, tableMtDynamoDbStreams)
+                Arguments.of(indexMtDynamoDb, indexMtDynamoDbStreams, indexMtDynamoDbStreamsCache),
+                Arguments.of(indexBinaryHkMtDynamoDb, indexBinaryHkMtDynamoDbStreams,
+                    indexBinaryHkMtDynamoDbStreamsCache),
+                Arguments.of(tableMtDynamoDb, tableMtDynamoDbStreams, null)
             );
         }
     }
 
+    /**
+     * Verifies describeStreamCache results match what's expected for a given key.
+     */
+    private void assertDescribeStreamCache(MtAmazonDynamoDbStreams mtDynamoDbStreams,
+                                           CachingAmazonDynamoDbStreams cachingStreams,
+                                           Stream stream,
+                                           boolean expectedCacheHit,
+                                           DescribeStreamResult expectedResult) {
+        // Setup
+        String key = stream.getStreamArn();
+        Cache<String, DescribeStreamResult> describeStreamCache = cachingStreams.getDescribeStreamCache();
+
+        // Verify cache hit/miss based on expected result
+        StreamsTestUtil.verifyDescribeStreamCacheResult(describeStreamCache, key, expectedCacheHit, expectedResult);
```

Comment:
Could we replace this method with a simple `assertEquals` on what is expected to be in the cache?

Reply:
Not in this case because there are only parts of the StreamDescription we care to verify so this leads to multiple assertions (streamArn, shards, streamStatus)

For example these `Stream Description` fields are not affected by the describeStream caching code:
StreamViewType: NEW_AND_OLD_IMAGES,
CreationRequestDateTime: Tue Jul 09 18:21:14 PDT 2019,
TableName: MtAmazonDynamoDbStreamsBaseTest.testDescribeStreamCache.SharedTable_b,
KeySchema: [{AttributeName: id,KeyType: HASH}],

The `MtAmazonDynamoDbStreamsBaseTest` test will have a stream description that has all of these fields set.  The `verifyDescribeStreamCacheResult` method will assert expected values match for these fields: streamArn, shards, streamStatus.

I also think adding the checks if the result is null (cache hit/miss) adds necessary coverage.